src/app/security/user-details.service.ts

```diff
@@ -105,4 +106,37 @@ export class UserDetailsService {
     localStorage.removeItem('dataSpecificationFolder');
     localStorage.removeItem('sdeAuthToken');
   }
+
+  setSdeResearchUser(user: ResearchUser) {
+    localStorage.setItem('sdeUserId', user.id);
+    localStorage.setItem('sdeEmail', user.email);
+    localStorage.setItem('sdePreferredName', user.preferredName ?? '');
+  }
+
+  getSdeResearchUser(): ResearchUser | null {
+    const userEmail = localStorage.getItem('sdeEmail');
+    if (!userEmail || userEmail.length === 0) {
+      return null;
+    }
+
+    return {
+      id: localStorage.getItem('sdeUserId') ?? '',
+      email: localStorage.getItem('sdeEmail') ?? '',
+      preferredName: localStorage.getItem('sdePreferredName') ?? '',
+      isDeleted: false,
```

Comment:
Is filling in all these fields with default values a sign of a problem elsewhere?

Reply:
This function wasn't strictly necessary so I've removed it instead.